L1Trigger/L1TMuonEndCap/interface/bdt/LossFunctions.h

```diff
@@ -0,0 +1,228 @@
+// LossFunctions.h
+// Here we define the different loss functions that can be used
+// with the BDT system. 
+
+#ifndef L1Trigger_L1TMuonEndCap_emtf_LossFunctions
+#define L1Trigger_L1TMuonEndCap_emtf_LossFunctions
+
+#include "Event.h"
+#include <string>
+#include <algorithm>
+
+// ========================================================
+// ================ Define the Interface ==================
+//=========================================================
+
+namespace emtf {
+
+// Define the Interface
+class LossFunction
+{
+    public:
+
+        // The gradient of the loss function.
+        // Each tree is a step in the direction of the gradient
+        // towards the minimum of the Loss Function.
+        virtual Double_t target(Event* e) = 0;
```

Comment:
please avoid root equivalents of standard types throughout. (Double_t -> double)

Reply:
Changed